aws_sagemaker_studio/sagemaker_studio_image_build/xgboost_bring_your_own/Batch_Transform_BYO_XGB.ipynb

```diff
@@ -0,0 +1,1112 @@
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Action\": [\n",
+    "                \"iam:GetRole\",\n",
+    "                \"iam:ListRoles\"\n",
+    "            ],\n",
+    "            \"Resource\": \"*\"\n",
+    "        },\n",
+    "        {\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Action\": \"iam:PassRole\",\n",
+    "            \"Resource\": \"arn:aws:iam::*:role/*\",\n",
+    "            \"Condition\": {\n",
+    "                \"StringLikeIfExists\": {\n",
+    "                    \"iam:PassedToService\": \"codebuild.amazonaws.com\"\n",
+    "                }\n",
+    "            }\n",
+    "        }\n",
+    "    ]\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's inspect the role we have created for our notebook here:\n",
+    "import boto3\n",
+    "import sagemaker\n",
+    "from sagemaker import get_execution_role\n",
+    "\n",
+    "role = get_execution_role()\n",
+    "sess = sagemaker.Session()\n",
+    "region = boto3.session.Session().region_name\n",
+    "print(\"Region = {}\".format(region))\n",
+    "sm = boto3.Session().client('sagemaker')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Check the SageMaker SDK Version. \n",
+    "\n",
+    "This notebook runs on SageMaker SDK Version < 2. Below we check the version, and if it is 2 or above, we reinstall the older version. If you see the \"Please restart the kernel\" prompt, simply click Kernel above and hit Restart. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "\n",
+    "if int(sagemaker.__version__.split('.')[0]) == 2:\n",
+    "    !{sys.executable} -m pip install sagemaker==1.72.0\n",
+    "    print(\"Installing previous SageMaker Version. Please restart the kernel\")\n",
+    "else:\n",
+    "    print(\"Version is good\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Restart if promted to do so above or else continue on"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Complete Setup: Import libraries and set global definitions.\n",
+    "\n",
+    "All needed libraries will come pre-installed with this notebook with the Lifecycle configuration scripts."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import os\n",
+    "from time import sleep, gmtime, strftime\n",
+    "import json\n",
+    "import time"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import SageMaker Experiments \n",
```

Comment:
Why does this example need experiments?

Reply:
Since this is an end to end example, it makes sense to cover experiments as a way to log training jobs and training metadata. This way we are consistent with the practices we are telling customers.